### DISCUSSION TITLE
Error while installing TPL

### DISCUSSION THREAD

**Discussion by hchen139** [sandialabs/seacas#416] [Q&A]
I'm following the steps outlined in the README.md to build SEACAS on my Macbook Pro machine. While installing TPL, I received the following error message:

Undefined symbols for architecture x86_64:
  "_SZ_BufftoBuffCompress", referenced from:
      _H5Z__filter_szip in H5Zszip.c.o
  "_SZ_BufftoBuffDecompress", referenced from:
      _H5Z__filter_szip in H5Zszip.c.o
  "_SZ_encoder_enabled", referenced from:
      _H5Z_init in H5Z.c.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)

Any help to resolve the issue will be much appreciated.

**Comment by hchen139** [comment]
The problem has been somewhat "solved" by setting "NEEDS_ZLIB=YES" and "NEEDS_SZIP=YES". The current issue relates to the getopt while running tests to check the installation:

The "getopt" executable that is available on this system is an older
version that is not compatible with the needs of the "decomp" tool.
If possible, you should update your getopt to a newer version and make
sure that the new getopt is in your path.

Below are some options for getting the current getopt version:
* If on a Mac: "sudo port install getopt" or "brew install gnu-getopt"
* Search the internet for "getopt-1.1.5" or "getopt-1.1.4"; download and build


The suggested options did not solve the issue. I'm running macOS Monterey 12.7.1.

**Comment by gsjaardema** [comment]
Re getopt:

Sorry, forgot to add this to the output that recommends using gnu-getopt from brew.
```
gnu-getopt is keg-only, which means it was not symlinked into /opt/homebrew,
because macOS already provides this software and installing another version in
parallel can cause all kinds of trouble.

If you need to have gnu-getopt first in your PATH, run:
  echo 'export PATH="/opt/homebrew/opt/gnu-getopt/bin:$PATH"' >> /Users/gdsjaar/.bash_profile
```
The important part is the last line with the `echo`

**Reply by hchen139** [reply]
By adding the above statement in the .bash_profile solved the problem. The new issue encountered is related to python. My computer has another package mamba installed that also provides the python package. And the MacOS also provide python3. Now, when I run SEACAS tests, it links  by default to the python provided by mamba instead of the MacOS. When run SEACAS tests, I got an error saying "ModuleNotFoundError: No module named 'numpy' " from the exodus.py file. Not sure whether this error message has anything to do with which correct python is used or not. Any idea how to solve this issue?

**Reply by gsjaardema** [reply]
The `numpy` package needs to be installed via `pip` or `pip3` and it provides numerical support.  I think it is possible to run exodus.py and exomerge.py without numpy support, so I should have the SEACAS tests determine whether numpy is installed and not try to use it if it isn't installed...

**Comment by gsjaardema** [comment]
> I'm following the steps outlined in the README.md to build SEACAS on my Macbook Pro machine. While installing TPL, I received the following error message:
> 
> Undefined symbols for architecture x86_64: "_SZ_BufftoBuffCompress", referenced from: _H5Z__filter_szip in H5Zszip.c.o "_SZ_BufftoBuffDecompress", referenced from: _H5Z__filter_szip in H5Zszip.c.o "_SZ_encoder_enabled", referenced from: _H5Z_init in H5Z.c.o ld: symbol(s) not found for architecture x86_64 clang: error: linker command failed with exit code 1 (use -v to see invocation)
> 
> Any help to resolve the issue will be much appreciated.
What part of the TPL build is giving the _SZ_* errors? Is that netCDF or in the HDF5 TPL build?

**Reply by hchen139** [reply]
It is HDF5. It didn't work with the zlib and szip installed using macports and homebrew. But the problem was solved by setting "NEEDS_ZLIB=YES" and "NEEDS_SZIP=YES" in install-tpl.sh.

**Reply by gsjaardema** [reply]
I'm glad that worked, but it shouldn't be needed.  My system builds without those defined.  I will see if I can figure out the difference and if I can, I will update the README or the install-tpl.
